qa-tests-backend/src/test/groovy/ProcessesListeningOnPortsTest.groovy

```diff
@@ -226,6 +238,104 @@ class ProcessesListeningOnPortsTest extends BaseSpecification {
 
         def list3 = processesListeningOnPorts.listeningEndpointsList
         assert list3.size() == 0
+
+        destroyDeployments()
+    }
+
+    @Tag("BAT")
+    def "Verify networking endpoints disappear when process is terminated"() {
+        given:
+        "When a deployment listening on a port is created and then the process is terminated"
+
+        setupSpec()
+        rebuildForRetries()
```

Comment:
I might lack some knowledge on the test lifecycle, but it looks to me that the second `rebuildForRetries` is redoing what the first one did? At least the deployment setup part. 

My understanding from what will happen with these two functions calls is:
1. Create test namespace
2. Create image pull secret
3. Create service account
4. Create deployments
5. Delete deployments
6. Create deployments

I'm probably missing something here. Is there something that makes `setupSpec` not run if the test is being retried but `rebuildForRetries` will?

Reply:
I have removed rebuildForRetries.